core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -0,0 +1,205 @@
+  private[kafka] case class Committed(offsets: Map[TopicPartition, OffsetAndMetadata])
+  //internal
+  private case object Poll
+
+  private var number = 0
+  private[kafka] def nextNumber() = {
+    this.synchronized {
+      number += 1
+      number
+    }
+  }
+  private[kafka] def rebalanceListener(onAssign: Iterable[TopicPartition] => Unit, onRevoke: Iterable[TopicPartition] => Unit) = new ConsumerRebalanceListener {
+    override def onPartitionsAssigned(partitions: util.Collection[TopicPartition]): Unit = {
+      onAssign(partitions.asScala)
+    }
+    override def onPartitionsRevoked(partitions: util.Collection[TopicPartition]): Unit = {
+      onRevoke(partitions.asScala)
+    }
+  }
+}
+
+private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V]) extends Actor {
+  import KafkaConsumerActor._
+
+  def pollTimeout() = settings.pollTimeout
+  def pollInterval() = settings.pollInterval
+
+  var requests = Map.empty[TopicPartition, ActorRef]
+  var consumer: KafkaConsumer[K, V] = _
+  var finished: Promise[Done] = _
+  var nextScheduledPoll: Option[Cancellable] = None
+  var pollExpected = false
+  var commitsInProgress = 0
+  var stopInProgress = false
+
+  def receive: Receive = LoggingReceive {
+    case Watch => sender() ! finished.future
+    case Assign(tps) =>
+      val previousAssigned = consumer.assignment()
+      consumer.assign((tps.toSeq ++ previousAssigned.asScala).asJava)
+    case AssignWithOffset(tps) =>
+      consumer.assign(tps.keys.toSeq.asJava)
+      tps.foreach {
+        case (tp, offset) => consumer.seek(tp, offset)
+      }
+    case Commit(offsets) =>
+      val commitMap = offsets.mapValues(new OffsetAndMetadata(_))
+      val reply = sender()
+      commitsInProgress += 1
+      consumer.commitAsync(commitMap.asJava, new OffsetCommitCallback {
+        override def onComplete(offsets: util.Map[TopicPartition, OffsetAndMetadata], exception: Exception): Unit = {
+          commitsInProgress -= 1
+          if (exception != null) reply ! Status.Failure(exception)
+          else reply ! Committed(offsets.asScala.toMap)
+        }
+      })
+      //right now we can not store commits in consumer - https://issues.apache.org/jira/browse/KAFKA-3412
+      pollExpected = true
+      poll()
+    case Subscribe(topics, listener) =>
+      consumer.subscribe(topics.toList.asJava, listener)
+    case SubscribePattern(pattern, listener) =>
+      consumer.subscribe(Pattern.compile(pattern), listener)
+    case Poll =>
+      pollExpected = true
+      poll()
+    case RequestMessages(topics) =>
+      requests ++= topics.map(_ -> sender()).toMap
+      pollExpected = true
+      poll()
+    case Stop =>
+      if (commitsInProgress == 0) {
+        context.stop(self)
+      }
+      else {
+        stopInProgress = true
+        context.become(stopping)
+      }
+  }
+
+  def stopping: Receive = LoggingReceive {
+    case Watch => sender() ! finished.future
+    case Assign(tps) =>
```

Comment:
Because we don't know how to react to confirmation or it absence and because these methods could not fail. These methods is just setup actions for client and will take effect only during next `poll`. I have no idea of any semantic with reply in such case.